rfcs/rfc-0000-discriminators.md

```diff
@@ -0,0 +1,208 @@
+
+  But we're approaching this from the standpoint that usernames on platforms such as these should not have to compete for 'uniqueness'.
+
+I want to take this section to also discuss the case study of Discord's removal of the discriminator system and their [cited drawbacks](https://discord.com/blog/usernames#heading-3):
+
+- > More than 40% of you either don’t remember your discriminator or don’t even know what a discriminator is. That’s a big problem when discriminators are required to add a new friend.
+
+  This RFC cannot address this issue however further work can be done to ease these interactions:
+
+  - R&D to determine how to build the UI to intuitively tell the user how to share their username with their discriminator
+  - Implementation of friend links / codes to ease connecting users to each other
+
+    This is not something that Discord has solved and to this day there is no way to create a friend request link.
+
+  - Implementation of nearby find (prior art: Discord) / QR friend codes (prior art: Snapchat) / add by contact book (prior art: Signal, WhatsApp)
+
+    These may help streamline IRL interactions by providing users with a simple flow to follow.
+
+  - Implementation of add by connections
+
+    Allow adding other people by common social media connections.
+
+  - Global user search (prior art: Steam)
+
+    Allow users to opt-in to a global username search / or otherwise also search through mutual members on servers, to avoid needing the discriminator altogether.
+
+- > Across Discord, almost half of all friend requests fail to connect the user with the person they wanted to match with, mostly because users enter an incorrect or invalid username due to a combination of missing discriminator and incorrect casing.
+
+  - Issues with ease of adding are addressed in the point above, however incorrect casing is irrelevant to Revolt as-is because Revolt's usernames are already case-insensitive and this will not change.
+
+    If we look at the given example:
+
+    > You meet someone IRL that you want to talk to on Discord, and they say “I’m Phibi Eight Nine Three Six!” You go home and add “phibi#8936” only to find out you added the wrong “Phibi” because your new friend’s username is actually “PhIBI#8936”.
+
+    Revolt does not permit a registration of both "PhIBI" and "phibi".
+
+- > You want to use a common name like “Mike” or “Jane” but there are already 9,999 Mikes or Janes so you’re blocked from that name altogether.
+
+  We are not restricted to just 4-digit discriminators in our implementation.
+
+- > You like to change your username a lot and get rate limited.
+
+  If we were also to add more stringent rate limits, this may be solved by also including display names.
+
+- > Your friend says they changed their name to “vernacular” but actually it’s “𝖛𝖊𝖗𝖓𝖆𝖈𝖚𝖑𝖆𝖗” and you have trouble finding them.
+
+  While this is a valid concern, I would personally put this down as the user's own issue.
```

Comment:
**user:** *has fancy special unicode in their username because it looks cool*
**their friends:** *can't add them easily*
**user:** :0

/j

but really i wonder why they don't just... add a warning that using uncommon unicode characters in your username might make it more difficult for people to find you.

Reply:
They also fail to mention that removing discriminators and forcing unique usernames is going to make this problem way, _way_ worse (since now it's basically a requirement to add random stuff to get that username.)
Also, I never read this blog post fully, how are the majority of these points not relating to discriminators at all?

Reply:
Adding a note here, it would be good to also warn users about weird and funky characters if there are any.

Reply:
@ggtylerr hate to go slightly off topic here, but holy shit it's so damn funny.

oh, you wanted the same username that 9,999 already have? well now only one person can ever have it! PROBLEM SOLVED!!! **WE DID IT REDDIT!!!!!!!**

Reply:
Added a note about this to the bottom of the RFC.